tests/integration/backward_compatible/sql_test.py

```diff
@@ -0,0 +1,590 @@
+        self._validate_rows(SqlColumnType.REAL, float)
+
+    def test_double(self):
+        self._populate_map_via_rc("new java.lang.Double(key)")
+        self._validate_rows(SqlColumnType.DOUBLE, float)
+
+    def test_date(self):
+        def value_factory(key):
+            return datetime.date(key + 2000, key + 1, key + 1)
+
+        self._populate_map_via_rc("java.time.LocalDate.of(key + 2000, key + 1, key + 1)")
+        self._validate_rows(SqlColumnType.DATE, value_factory)
+
+    def test_time(self):
+        def value_factory(key):
+            return datetime.time(key, key, key, key)
+
+        self._populate_map_via_rc("java.time.LocalTime.of(key, key, key, key * 1000)")
+        self._validate_rows(SqlColumnType.TIME, value_factory)
+
+    def test_timestamp(self):
+        def value_factory(key):
+            return datetime.datetime(key + 2000, key + 1, key + 1, key, key, key, key)
+
+        self._populate_map_via_rc(
+            "java.time.LocalDateTime.of(key + 2000, key + 1, key + 1, key, key, key, key * 1000)"
+        )
+        self._validate_rows(SqlColumnType.TIMESTAMP, value_factory)
+
+    def test_timestamp_with_time_zone(self):
+        def value_factory(key):
+            return datetime.datetime(
+                key + 2000,
+                key + 1,
+                key + 1,
+                key,
+                key,
+                key,
+                key,
+                timezone(datetime.timedelta(hours=key)),
+            )
+
+        self._populate_map_via_rc(
+            "java.time.OffsetDateTime.of(key + 2000, key + 1, key + 1, key, key, key, key * 1000, "
+            "java.time.ZoneOffset.ofHours(key))"
+        )
+        self._validate_rows(SqlColumnType.TIMESTAMP_WITH_TIME_ZONE, value_factory)
+
+    def test_decimal(self):
+        def value_factory(key):
+            return decimal.Decimal((0, tuple(map(int, str(abs(key)))), -1 * key))
+
+        self._populate_map_via_rc(
+            'new java.math.BigDecimal(new java.math.BigInteger("" + key), key)'
+        )
+        self._validate_rows(SqlColumnType.DECIMAL, value_factory)
+
+    def test_null(self):
+        self._populate_map()
+        result = self.client.sql.execute("SELECT __key, NULL AS this FROM %s" % self.map_name)
+        self._validate_result(result, SqlColumnType.NULL, lambda _: None)
+
+    def test_object(self):
+        def value_factory(key):
+            return Student(key, key)
+
+        self._populate_map(value_factory=value_factory)
+        result = self.client.sql.execute("SELECT __key, this FROM %s" % self.map_name)
+        self._validate_result(result, SqlColumnType.OBJECT, value_factory)
+
+    def test_null_only_column(self):
+        self._populate_map()
+        result = self.client.sql.execute(
+            "SELECT __key, CAST(NULL AS INTEGER) FROM %s" % self.map_name
+        )
+        self._validate_result(result, SqlColumnType.INTEGER, lambda _: None)
```

Comment:
For code coverage

Reply:
Which line(s) does this test cover, but not any other test? I think `CAST(NULL AS INTEGER)` is weird and probably won't be used at all. I don't think we should add tests just for coverage

Reply:
See `ListCNFixedSizeCodec`

```py
        if type == ListCNFixedSizeCodec._TYPE_NULL_ONLY:
            return [None] * count
```

There is no way I could find to test the `else:` branch with the server though, in the same codec for IMDG 4.2.

Reply:
How can someone add a null value to a map? I think it is impossible with map methods. I think due to this reason you used `CAST(NULL AS INTEGER)`, which I find weird. If a way is introduced in the future to add null values to a map via SQL, I would rather change this test later.

Reply:
This is how the fixed-size columns are encoded in Java, so I assume it should be possible now (or in the future), and clients should work in the future. So, I find a way to test some part of it with IMDG 4.2, and I think this is valuable. I also spent a considerable amount of time to get a response for the `else` part with 4.2, but couldn't succeeded. So, I would test it now( as long as I can)

Reply:
I think your approach is a workaround, since no null data is not stored in a map, but you add an extra expression to your query. With SQL we are querying map data, that's why I mentioned adding null values to a map.

Reply:
Yes, I know it is a workaround. Like I said, Java encodes columns like this and on the client-side, I try to decode it. Maybe in the future, we will add this feature and I would like the client to work correctly, even now. 

So, I am not thinking about removing this. As I said, if you can find a workaround to test the `else` part too, I could happily add a test for it